tests/e2e-client/src/test/java/org/glassfish/jersey/tests/e2e/client/connector/ssl/SslConnectorHostnameVerifierTest.java

```diff
@@ -66,32 +67,40 @@ protected String clientTrustStore() {
     @ParameterizedTest
     @MethodSource("testData")
     public void testHostnameVerifierApplied(ConnectorProvider connectorProvider) throws Exception {
-        // Grizzly and Jetty connectors don't support Hostname Verification
-        if (isExcluded(Arrays.asList(GrizzlyConnectorProvider.class, JettyConnectorProvider.class), connectorProvider)) {
+        // Grizzly connector does not support Hostname Verification
+        if (isExcluded(Arrays.asList(GrizzlyConnectorProvider.class), connectorProvider)) {
             return;
         }
 
         final Client client = ClientBuilder.newBuilder()
-                .withConfig(new ClientConfig().connectorProvider(connectorProvider))
+                .withConfig(getClientConfig(JettyConnectorProvider.class.isAssignableFrom(connectorProvider.getClass()))
+                        .connectorProvider(connectorProvider))
                 .register(HttpAuthenticationFeature.basic("user", "password"))
                 .hostnameVerifier(new CustomHostnameVerifier())
                 .sslContext(getSslContext())
                 .build();
 
         try {
             client.target(Server.BASE_URI).request().get(Response.class);
-            fail("HostnameVerifier was not applied.");
+            fail("HostnameVerifier was not applied by " + connectorProvider.getClass());
         } catch (ProcessingException pex) {
             CustomHostnameVerifier.HostnameVerifierException hve = getHVE(pex);
 
             if (hve != null) {
                 assertEquals(CustomHostnameVerifier.EX_VERIFIER_MESSAGE, hve.getMessage());
             } else {
-                fail("Invalid wrapped exception.");
+                fail("Invalid wrapped exception.", pex);
             }
         }
     }
 
+    private static final ClientConfig getClientConfig(boolean enableSslHostnameVerification) {
+        final ClientConfig config = new ClientConfig();
+        return enableSslHostnameVerification ? config
+                .property(JettyClientProperties.ENABLE_SSL_HOSTNAME_VERIFICATION, Boolean.FALSE)
```

Comment:
Or the other way around. Does the user need to set `property(JettyClientProperties.ENABLE_SSL_HOSTNAME_VERIFICATION, Boolean.FALSE)` when the custom hostnameverifier is registered? Maybe it can be set to false automatically, can it not?